### ISSUE TITLE
.renvignore doesn't ignore files when checking project synchronicity

### ISSUE THREAD

**Issue by 3styleJam** [rstudio/renv#2033]
Hi. I am having difficulties with implementing .renvignore within a github repository. There is a .gitignore file at the root, and another .gitignore file within the "renv" folder. There are two folders to ignore, both named "archive" and are located in two different places (Apologies I don't know how to make a nice looking repository tree):

root/.gitignore.txt
root/archive
root/archive/script_1.R
root/subfolder
root/subfolder/archive
root/subfolder/archive/script_2.R
root/renv
root/renv/.gitignore.txt

I performed the following:

1. Remove both archive folders temporarily from the repository, since my understanding is git is already tracking them and .gitignore ensures only untracked files/folders remain untracked. Assuming .renvignore works in a similar way, I thought best to temporarily remove them.
2. Add the .renvignore file with the line: `**/archive/` . Also, root/.gitignore.txt contains a line that ignores all text files: `*.txt` , so I've had to add either `!.renvignore.txt` or `!renv/.renvignore.txt` to be able to commit .renvignore to my repository. 
3. Added the archive folders back into my repository
4. Opened "root/archive/script_1.R" in RStudio. On startup, the console reads "Project ... loaded." suggesting my project is in sync. Then I added a line to my script: `library(<package not currently used in project>)` and restarted RStudio.

Based on my current understanding, I expected the console to read "Project ... loaded." again as .renvignore says to ignore that file when looking for used packages, however instead it says "The project is out-of-sync -- use `renv::status()` for details." and upon running renv::status(), it identifies the package from the line I added to the script and says it is not installed.

I have tried several different methods but all result in .renvignore seemingly not working. I am not sure whether renv is even recognising the .renvignore file and I don't know how I would check. Any help with identifying the cause of the problem and fixing it would be much appreciated, thank you.

**Comment by 3styleJam** [comment]
> Any chance you can share a reproducible example?
> 
> If I'm understanding correctly, I don't think you should need to include the root directory in your `.renvignore`; it should suffice to just have:
> 
> ```
> archive
> subdir/archive
> ```
> 
> or something like that.

Sorry, I made it confusing. Where I've put "root/<file/subdir>" above, I just used that to indicate that the file/subdir is located within the root directory.

I am unsure how to create a good reprex for this, but you could try following these steps:

1. Create a new private github repository e.g. "Repository-X".
2. Add a new R project from RStudio. This create a new .Rproj and .gitignore file.
3. Make sure the following lines are present in the .gitignore file:
```
*.txt
!/renvignore.txt
``` 
4. Install renv and create an empty project library:
```r
install.packages("renv")
renv::init()
``` 
If successful, should have the "renv" folder, .renv lockfile and a .Rprofile .
5. Add "10_archive/script_1.R" with the line `library(tidyverse)`
6. Call `renv::snapshot()` . Should identify package(s) within the project, choose option 2 in the prompt to install them.
7. Add ".renvignore.txt" with the line `subdir/10_archive/` . 
8. Add "subdir/10_archive/script_2" with the line `library(biopixR)` (or another package(s) not yet installed).
9. Restart RStudio. The console shows:
```r
#> The project is out-of-sync -- use `renv::status()` for details.

renv::status()
#> The following package(s) are used in this project, but are not installed:
#> - biopixR
#> 
#> See renv::status for advice on resolving these issues.
``` 

This is unexpected. I expect the project to be in sync because `library(biopixR)` is within a script within a subdirectory that should be ignored according to the contents of .renvignore .

**Comment by kevinushey** [comment]
The file name should just be `.renvignore`, not `.renvignore.txt` -- could that be the cause of the issue you're seeing?

**Comment by 3styleJam** [comment]
> The file name should just be `.renvignore`, not `.renvignore.txt` -- could that be the cause of the issue you're seeing?

Ahh yes, this was the problem. I had used Notepad to create the .renvignore file so it was treated as a text file and not a RENVIGNORE file, which is how it should appear in Windows File Explorer under Type. Using Git Bash and using the follow command creates the file correctly:

```
$ cat > .renvignore
subdir/10_archive
``` 

I had wrongly assumed it would look like .gitignore in File Explorer which does have Type of "Text Document", however that file was generated when initialising renv. 

Thanks very much for your help. Closing this issue now as problem is resolved.
